xapian-applications/omega/worker.h

```diff
@@ -0,0 +1,87 @@
+/** @file worker.h
+ * @brief Class representing worker process.
+ */
+/* Copyright (C) 2011 Olly Betts
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <cstdio>
+#include <string>
+#include <sys/types.h>
+
+/** An object to communicate with the assistant process
+ *
+ *  It is possible that an external library contain errors that can cause omindex
+ *  termination or blocking indexing. For that reason, it is used a subprocess
+ *  'assistant' that use the external library and communicate the results. This
+ *  way, library bugs are isolated and they cannot damage omindex.
+ *
+ *  Each worker is associated to a particular assistant.
+ *
+*/
+class Worker {
+    /// @private Workers ignore SIGPIPE.
+    static bool ignoring_sigpipe;
+    /// @private PID of the assistant process.
+    pid_t child;
+    /** @private Socket for supporting communication between the worker
+     *	and its assistant.
+    */
+    std::FILE * sockt;
+    /// @private Name of the assistant program.
+    std::string filter_module;
+    /// @private This method create the assistant subprocess.
```

Comment:
You don't need `@private` here - these are `private` (that's the default for `class` before any access specifier) and doxygen knows that.  You only need to say `@private` is you want to tell doxygen to treat something as having a different access to what it actually has (e.g. in the xapian-core API we sometimes make the `internal` member public to avoid having to forward declare internal classes in the public API header, but we want users to treat it as if it is private so we use `@private` so doxygen doesn't show it as being public.)

Reply:
Okay, I am going to fix it. I believed that I need to told doxygen that it was private.

Reply:
Actually you do for some languages (which don't have the concept in the language itself), but not for C++:

http://www.doxygen.nl/manual/commands.html#cmdprivate says:

> Note that doxygen automatically detects the protection level of members in object-oriented languages. This command is intended for use only when the language does not support the concept of protection level natively (e.g. C, PHP 4).